### DISCUSSION TITLE
在windows的QT的环境请问如何在浏览器执行js代码，然后返回执行的结果（回调）到CPP？

### DISCUSSION THREAD

**Discussion by messigoat3339** [CefView/QCefView#281] [Q&A] [answered]
你好
在qwebengineview中，只执行js代码是 `QWebEngineView()->page()->runJavaScript(js_code.c_str());`
如果需要在执行完后返回结果是
```
ui->webEngineView->page()->runJavaScript("get_something();", [this,参数...](const QVariant &v)
		{
			const QString result_code = v.toString();

		});
```

请问在QCefView有类似的方法吗？还是用到CefViewQuery的例子，先执行runJavaScript，然后在js代码中异步调用，反馈给cpp响应?

**Comment by L-Super** [comment]
有的，请参见文档：https://cefview.github.io/QCefView/zh/docs/guide/Interoperability

**Comment by tishion** [comment, marked as answer]
/// <summary>
  /// Gets called on the result of the javascript executed with <see cref="executeJavascriptWithResult"/> returned
  /// </summary>
  /// <param name="browserId">The browser id</param>
  /// <param name="frameId">The frame id</param>
  /// <param name="context">The context</param>
  /// <param name="result">The result</param>
  void reportJavascriptResult(int browserId, qint64 frameId, qint64 context, const QVariant& result);

看一下demo里是如何使用这个signal的
